### PR TITLE
[Dashboard] feat: replace all teams filter in remaining pages

### LINKED ISSUE #2860: [Dashboard] Update colony teams navigation on remaining pages
### Story

As a USER I want to easily switch between teams in the dashboard so that only team-relevant information is displayed.

### Description

After the **[Dashboard] Update colony teams navigation** is completed, please replace the breadcrumbs navigation on the remaining pages where we display domain filtering

**Before**
Image

**After**
Image

## Implementation
Maybe we need to use a context to store the active domain or just a path parameter

### This issue is complete when...

- [ ] Selecting one team navigation tab activates domain filtering

[Figma link](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4859-36367&t=3YmH0OCdhMXflaYC-0)
[Figma link empty state](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4867-40639&t=3YmH0OCdhMXflaYC-0)

### PR DESCRIPTION
**Pull request by bassgeta**
## Description

Replace the old breadcrumb navigation with the new filter component!
[Figma link](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4870-44099&t=pWU79XSFvRIH7LiK-4)
image

## Testing

Basically go to the following pages and make sure that the filter still works and that it displays nicely
1. Activity feed
2. Balances
3. Incoming funds
4. Teams
5. Agreements
6. Members
7. Permissions

And do note there's one layout bug present but it's being fixed elsewhere
image

When the screen gets wide, the filter is stuck with the content, but the title stays on the lefthand side.

## Diffs

**Changes** 🏗

* Added the `TeamFilter` component to the 7 pages above 

**Deletions** ⚰️

* `BreadcrumbsCardSelect` component
* `useTeamsBreadcrumbs` hook

Resolves  #2860

### CODE REVIEW

File: src/components/frame/v5/pages/ActivityPage/ActivityPage.tsx [hunk 1]
```diff
@@ -2,28 +2,25 @@ import React, { type FC } from 'react';
...
   const widgets = useActivityFeedWidgets();
 
   return (
     <div className="flex w-full flex-col gap-4 sm:gap-6">
+      <div className="mb-2">
+        <TeamFilter />
+      </div>
       <WidgetBoxList items={widgets} />
       <div>
         <FiltersContextProvider>
```
Comment (mmioana) [inline src/components/frame/v5/pages/ActivityPage/ActivityPage.tsx:26]: For consistency, would suggest to wrap the previous content as you did on the previous pages
```
<div className="flex flex-col gap-8">
      <TeamFilter />
     ...previous page content here
</div>
```

File: src/components/frame/v5/pages/AgreementsPage/AgreementsPage.tsx [hunk 2]
```diff
@@ -54,6 +49,9 @@ const AgreementsPage: FC = () => {
 
   return (
     <div>
```
Comment (mmioana) [inline src/components/frame/v5/pages/AgreementsPage/AgreementsPage.tsx:51]: Here as well

```
<div className="flex flex-col gap-8">
      <TeamFilter />
     ...previous page content here
</div>
```

File: src/components/frame/v5/pages/MembersPage/MembersPage.tsx [hunk 3]
```diff
@@ -98,70 +99,73 @@ const MembersPage: FC = () => {
...
-              titleAction={titleAction}
-            >
-              <Outlet />
-            </MembersTabContentWrapper>
-          ),
-        },
-      ]}
-    />
+    <div className="flex flex-col gap-8">
+      <TeamFilter />
```
Comment (mmioana) [inline src/components/frame/v5/pages/MembersPage/MembersPage.tsx:103]: Here as well

```
<div className="flex flex-col gap-8">
      <TeamFilter />
     ...previous page content here
</div>
```

File: src/components/frame/v5/pages/TeamsPage/TeamsPage.tsx [hunk 4]
```diff
@@ -35,6 +36,9 @@ const TeamsPage: FC = () => {
 
   return (
     <div className="flex w-full flex-col gap-6">
+      <div className="mb-2">
+        <TeamFilter />
+      </div>
```
Comment (mmioana) [inline src/components/frame/v5/pages/TeamsPage/TeamsPage.tsx:41]: Here as well

```
<div className="flex flex-col gap-8">
      <TeamFilter />
     ...previous page content here
</div>
```

**Review by mmioana — CHANGES_REQUESTED** [review]
Smashed another one @bassgeta 💪 👏 

Tested and can confirm the new filters are present on the mentioned pages
Screenshot 2024-09-18 at 11 21 12
Screenshot 2024-09-18 at 11 21 21
Screenshot 2024-09-18 at 11 21 35
Screenshot 2024-09-18 at 11 21 39
Screenshot 2024-09-18 at 11 21 45
Screenshot 2024-09-18 at 11 21 51
Screenshot 2024-09-18 at 11 21 58
Screenshot 2024-09-18 at 11 22 03

I know you didn't touch these pages, but can confirm the domain filters is not present on the Extensions or Profile pages
Screenshot 2024-09-18 at 11 22 10
Screenshot 2024-09-18 at 11 22 20

After leaving the refactoring comments for consistency, I'm wondering if it wouldn't make sense to create a new component and just wrap the pages needing the domain filter. What do you think @bassgeta? 😁 

```
const ContentWithDomainFilter: FC<PropsWithChildren> = ({ children }) => {

  return (
    <div className="flex flex-col gap-8">
      <TeamFilter />
      {children}
    </div>
  )
}
```

**Review by mmioana — APPROVED** [review]
Thank you for implementing the suggested changes @bassgeta 💪 💯 

Screenshot 2024-09-19 at 13 40 33
Screenshot 2024-09-19 at 13 40 41
Screenshot 2024-09-19 at 13 40 44
Screenshot 2024-09-19 at 13 40 46
Screenshot 2024-09-19 at 13 40 50
Screenshot 2024-09-19 at 13 40 53
Screenshot 2024-09-19 at 13 40 55
Screenshot 2024-09-19 at 13 40 58

Let's get this one merged 🎉

**Review by davecreaser — APPROVED** [review]
Nice job once again @bassgeta!

It's nice that now the breadcrumbs can just be breadcrumbs 😆 It makes it much simpler! Also agree with the approach of the reusable wrapper component, it's very nice.

It looks like you need to rebase, but I checked the code by just looking at your actual commits and it all looks good. Also tested all the pages you mentioned (including mobile) and everything is as expected! Let's gooo.

Screenshot 2024-09-19 at 17 12 37
Screenshot 2024-09-19 at 17 12 53
Screenshot 2024-09-19 at 17 13 07
Screenshot 2024-09-19 at 17 13 18
Screenshot 2024-09-19 at 17 13 30
Screenshot 2024-09-19 at 17 13 41
Screenshot 2024-09-19 at 17 13 53

**Review by rdig — APPROVED** [review]
Very nice team filters. Tested all pages and all seem to be behaving correctly. Stellar work!

[Screencast from 2024-09-20 12-30-38.webm](https://github.com/user-attachments/assets/553c0b3f-6473-4dc1-bce7-9be67707f8ef)

**Pull request merged**